api/handler/check-surveys.js

```diff
@@ -41,18 +43,24 @@ function checkSurveys (request, reply) {
         .then(() => {
             database.sequelize.query(
                 `
-                SELECT *
-                FROM survey_instance AS si
+                SELECT *, si.id
+                FROM survey_instance si
                 JOIN patient pa
                 ON si.patientId = pa.id
                 JOIN survey_template st
                 ON si.surveyTemplateId = st.id
                 WHERE pa.pin = ?
+                AND si.startTime <= ?
+                AND si.endTime > ?
+                AND si.surveyInstanceCompleted = ?
                 `,
                 {
                     type: database.sequelize.QueryTypes.SELECT,
                     replacements: [
-                        currentPatient.pin
+                        currentPatient.pin,
+                        currentDate.toISOString(),
+                        currentDate.toISOString(),
+                        surveyCompletedStatus
```

Comment:
Yeah. I can pass it directly in the query. Will change this.
On Nov 19, 2015 1:34 PM, "Christian Murphy" <notifications@github.com>
wrote:

> In api/handler/check-surveys.js
> <https://github.com/ser515asu/PRP-Manhattan-Project/pull/106#discussion_r45396468>
> :
>
> >                  `,
> >                  {
> >                      type: database.sequelize.QueryTypes.SELECT,
> >                      replacements: [
> > -                        currentPatient.pin
> > +                        currentPatient.pin,
> > +                        currentDate.toISOString(),
> > +                        currentDate.toISOString(),
> > +                        surveyCompletedStatus
>
> Why is there a replacement for a hard coded constant value?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ser515asu/PRP-Manhattan-Project/pull/106/files#r45396468>
> .
>